BrainPortal/spec/models/tool_spec.rb

```diff
@@ -23,14 +23,105 @@
 require 'rails_helper'
 
 describe Tool do
-  let(:tool) { build(:tool) }
+  let(:tool) do
+    allow(File).to receive(:exist?).and_return(true)
+    allow(File).to receive(:exist?).with(Rails.root + "public/licenses/bad.html").and_return(false)
```

Comment:
I still find the rspec a bit mysterious beast. Yet 1) It literally told me to swap the order of statements when I tried a different order. 2) if I replace "bad" with "good" the affected tests would fail, so looks like the latest mock has sort of higher priority.

Reply:
The latest allow seem have priority https://github.com/rspec/rspec-mocks/issues/557

Reply:
Excellent. Just checking.